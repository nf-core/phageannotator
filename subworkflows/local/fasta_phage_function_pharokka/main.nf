//
// Functionally annotate phage sequences using Pharokka
//

include { PHAROKKA_INSTALLDATABASES } from '../../../modules/nf-core/pharokka/installdatabases/main'
include { PHAROKKA_PHAROKKA         } from '../../../modules/nf-core/pharokka/pharokka/main'

workflow FASTA_PHAGE_FUNCTION_PHAROKKA {
    take:
    virus_fasta     // [ [ meta ], fasta ]  , virus sequences (mandatory)
    pharokka_db     // [ pharokka_db ]      , Pharokka database directory (optional)

    main:
    ch_versions = Channel.empty()

    // if pharokka_db exists, skip database download
    if ( pharokka_db ){
        ch_pharokka_db = pharokka_db
    } else {
        //
        // MODULE: download standard pharokka database
        //
        ch_pharokka_db = PHAROKKA_INSTALLDATABASES( ).pharokka_db
        ch_versions = ch_versions.mix(PHAROKKA_INSTALLDATABASES.out.versions)
    }

    //
    // MODULE: Functionally annotate phages
    //
    ch_pharokka_gbk_gz = PHAROKKA_PHAROKKA ( virus_fasta, ch_pharokka_db ).gbk
    pharokka_final_output_tsv = PHAROKKA_PHAROKKA.out.cds_final_merged_output
    ch_versions = ch_versions.mix(PHAROKKA_PHAROKKA.out.versions.first())

    emit:
    versions                    = ch_versions               // [ versions.yml ]
    pharokka_gbk_gz             = ch_pharokka_gbk_gz        // [ [ meta ], gbk.gz ]
    pharokka_final_output_tsv   = pharokka_final_output_tsv // [ [ meta ], tsv ]
}
