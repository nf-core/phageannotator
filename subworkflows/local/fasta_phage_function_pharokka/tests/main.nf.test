nextflow_workflow {

    name "Test Subworkflow: FASTA_PHAGE_FUNCTION_PHAROKKA"
    script "../main.nf"
    workflow "FASTA_PHAGE_FUNCTION_PHAROKKA"
    config "./nextflow.config"

    // Dependencies
    tag "PHAROKKA_INSTALLDATABASES"
    tag "PHAROKKA_PHAROKKA"


    test("fasta") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ]
                )
                input[1] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.versions,
                    workflow.out.pharokka_final_output_tsv,
                    path(workflow.out.pharokka_gbk_gz[0][1]).linesGzip[1..50]
                    ).match()
                }
            )
        }
    }
}
