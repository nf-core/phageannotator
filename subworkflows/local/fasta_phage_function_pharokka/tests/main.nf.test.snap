{
    "fasta": {
        "content": [
            [
                "versions.yml:md5,7715e330cd5621f95141e256b567ec19",
                "versions.yml:md5,b225eb2829469045a5bda2c5d7a69009"
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_cds_final_merged_output.tsv:md5,1242b6ffe6e53ef5bce0a9a2cac3cd45"
                ]
            ],
            [
                "DEFINITION  MT192765.1 Severe acute respiratory syndrome coronavirus 2 isolate",
                "            SARS-CoV-2/human/USA/PC00101P/2020, complete genome.",
                "ACCESSION   MT192765",
                "VERSION     MT192765.1",
                "KEYWORDS    .",
                "SOURCE      .",
                "  ORGANISM  .",
                "            .",
                "FEATURES             Location/Qualifiers",
                "     CDS             1..129",
                "                     /ID=\"TEST_CDS_0001\"",
                "                     /transl_table=11",
                "                     /phrog=\"No_PHROG\"",
                "                     /top_hit=\"No_PHROG\"",
                "                     /locus_tag=\"TEST_CDS_0001\"",
                "                     /function=\"unknown function\"",
                "                     /product=\"hypothetical protein\"",
                "                     /source=\"PHANOTATE_1.5.1\"",
                "                     /score=\"-3.2955849959257986\"",
                "                     /phase=\"0\"",
                "                     /translation=\"VYTFPGNKPTNFRSLVDLFSKRTLKSVWLSLGCMLSALTQYN\"",
                "     CDS             259..13476",
                "                     /ID=\"TEST_CDS_0002\"",
                "                     /transl_table=11",
                "                     /phrog=\"No_PHROG\"",
                "                     /top_hit=\"No_PHROG\"",
                "                     /locus_tag=\"TEST_CDS_0002\"",
                "                     /function=\"unknown function\"",
                "                     /product=\"hypothetical protein\"",
                "                     /source=\"PHANOTATE_1.5.1\"",
                "                     /score=\"-9.626540230974856e+92\"",
                "                     /phase=\"0\"",
                "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQH",
                "                     LKDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGETL",
                "                     GVLVPHVGEIPVAYRKVLLRKNGNKGAGGHSYGADLKSFDLGDELGTDPYEDFQENWNT",
                "                     KHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQLDFI",
                "                     DTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDTFNGECPNFVFPLNSII",
                "                     KTIQPRVEKKKLDGFMGRIRSVYPVASPNECNQMCLSTLMKCDHCGETSWQTGDFVKAT",
                "                     CEFCGTENLTKEGATTCGYLPQNAVVKIYCPACHNSEVGPEHSLAEYHNESGLKTILRK",
                "                     GGRTIAFGGCVFSYVGCHNKCAYWVPRASANIGCNHTGVVGEGSEGLNDNLLEILQKEK",
                "                     VNINIVGDFKLNEEIAIILASFSASTSAFVETVKGLDYKAFKQIVESCGNFKVTKGKAK",
                "                     KGAWNIGEQKSILSPLYAFASEAARVVRSIFSRTLETAQNSVRVLQKAAITILDGISQY",
                "                     SLRLIDAMMFTSDLATNNLVVMAYITGGVVQLTSQWLTNIFGTVYEKLKPVLDWLEEKF",
                "                     KEGVEFLRDGWEIVKFISTCACEIVGGQIVTCAKEIKESVQTFFKLVNKFLALCADSII",
                "                     IGGAKLKALNLGETFVTHSKGLYRKCVKSREETGLLMPLKAPKEIIFLEGETLPTEVLT",
                "                     EEVVLKTGDLQPLEQPTSEAVEAPLVGTPVCINGLMLLEIKDTEKYCALAPNMMVTNNT",
                "                     FTLKGGAPTKVTFGDDTVIEVQGYKSVNITFELDERIDKVLNEKCSAYTVELGTEVNEF",
                "                     ACVVADAVIKTLQPVSELLTPLGIDLDEWSMATYYLFDESGEFKLASHMYCSFYPPDED",
                "                     EEEGDCEEEEFEPSTQYEYGTEDDYQGKPLEFGATSAALQPEEEQEEDWLDDDSQQTVG",
                "                     QQDGSEDNQTTTIQTIVEVQPQLEMELTPVVQTIEVNSFSGYLKLTDNVYIKNADIVEE"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T04:53:13.97160193"
    }
}